python-package/lightgbm/basic.py

```diff
@@ -620,6 +620,9 @@ def _lazy_init(self, data, label=None, max_bin=255, reference=None,
         if data is None:
             self.handle = None
             return
+        if reference is not None:
+            self.pandas_categorical = reference.pandas_categorical
+            categorical_feature = reference.categorical_feature
```

Comment:
@wxchan here had a bug when using validation with categorical features. 
Can you check other places that may have the same bug ?